apps/settings/js/modules/bluetooth/bluetooth_context.js

```diff
@@ -0,0 +1,852 @@
+        }
+
+        // Update state
+        this.state = state;
+        this.debug('_updateStatus(): set state = ' + state);
+
+        // Update enabled
+        this.enabled = enabled;
+        this.debug('_updateStatus(): set enabled = ' + enabled);
+      });
+    },
+
+    /**
+     * Set Bluetooth enable/disable.
+     *
+     * @access public
+     * @memberOf BluetoothContext
+     * @param {Boolean} enabled
+     */
+    setEnabled: function btc_setEnabled(enabled) {
+      this.debug('setEnabled(): request set enabled = ' + enabled);
+      if ((this.enabled === enabled) ||
+          (this.state === 'enabling') ||
+          (this.state === 'disabling')) {
+        return;
+      }
+
+      if (!this._defaultAdapter) {
+        return;
+      }
+
+      if (enabled) {
+        this._defaultAdapter.enable().then(() => {
+          this.debug('setEnabled(): set enable successfully :)');
+          this._updateStatus('enabled');
+        }, (reason) => {
+          this.debug('setEnabled(): set enable failed: reason = ' + reason);
+        });
+      } else {
+        this._defaultAdapter.disable().then(() => {
+          this.debug('setEnabled(): set disable successfully :)');
+          this._updateStatus('disabled');
+        }, (reason) => {
+          this.debug('setEnabled(): set disable failed: reason = ' + reason);
+        });
+      }
+    },
+
+    /**
+     * Set Bluetooth discoverable.
+     *
+     * @access public
+     * @memberOf BluetoothContext
+     * @param {Boolean} enabled
+     */
+    setDiscoverable: function btc_setDiscoverable(enabled) {
+      if ((this.discoverable === enabled) || (this.state !== 'enabled')) {
+        return;
+      }
+
+      if (!this._defaultAdapter) {
+        return;
+      }
+
+      this._defaultAdapter.setDiscoverable(enabled).then(() => {
+        this.debug('setDiscoverable(): set discoverable successfully :)');
+        this.discoverable = enabled;
+      }, (reason) => {
+        this.debug('setDiscoverable(): set discoverable failed: ' + 
+                   'reason = ' + reason);
+      });
+    },
+
+    /**
+     * Set adapter name.
+     *
+     * @access public
+     * @memberOf BluetoothContext
+     * @param {String} name
+     */
+    setName: function btc_setName(name) {
+      if (name === this.name) {
+        return;
+      }
+
+      if (!this._defaultAdapter) {
+        return;
+      }
+
+      this._defaultAdapter.setName(name).then(() => {
+        this.debug('setName(): set name successfully :) name = ' +
+                    this._defaultAdapter.name);
+        this.name = this._defaultAdapter.name;
+      }, (reason) => {
+        this.debug('setName(): set name failed: reason = ' + reason);
+      });
+    },
+
+    /**
+     * Set adapter name by product model.
+     *
+     * @access public
+     * @memberOf BluetoothContext
+     */
+    setNameByProductModel: function btc_setNameByProductModel() {
```

Comment:
It's using for setting an empty name from `bluetooth/panel.js`. It provides a function while a user wants to set a empty name. Then, we assign product model to be the name. Sync user story currently.